MapboxCoreNavigation/RouteController.swift

```diff
@@ -598,6 +575,32 @@ extension RouteController: CLLocationManagerDelegate {
         checkForFasterRoute(from: location)
     }
     
+    func monitorLegProgress(_ location: CLLocation) {
+        guard routeProgress.currentLegProgress.upComingStep?.maneuverType == .arrive else { return }
+        
+        let polyline = Polyline(routeProgress.currentLegProgress.currentStep.coordinates!)
+        let userSnapToStepDistanceFromManeuver = polyline.distance(from: location.coordinate)
+        let secondsToEndOfStep = userSnapToStepDistanceFromManeuver / location.speed
+        guard secondsToEndOfStep < RouteControllerDurationRemainingWaypointArrival else { return }
+        
+        let currentDestinationWaypoint = routeProgress.currentLeg.destination
+        guard currentDestinationWaypoint != previousArrivalWaypoint else { return }
+        
+        routeProgress.currentLegProgress.userHasArrivedAtWaypoint = true
+        previousArrivalWaypoint = currentDestinationWaypoint
+        delegate?.routeController?(self, didArriveAt: routeProgress.currentLeg.destination)
+            
+        if let lastRouteWaypoint = routeProgress.route.legs.last?.destination, lastRouteWaypoint == currentDestinationWaypoint {
+            sessionState.arrivalTimestamp = Date()
+            sendArriveEvent()
+        }
+        
+        if !routeProgress.isFinalLeg,
+            (delegate?.routeController?(self, shouldIncrementLegWhenArrivingAtWaypoint: routeProgress.currentLeg.destination) ?? true) {
```

Comment:
I’d still like for this delegate method to go away (#1009), either as part of this PR or as part of #1010. I was thinking that `routeController(_:didArriveAt:)` would return a Boolean, which would be stored in a local variable above and fed into this check. Do you see any problems with that approach?

Reply:
OK, let’s do this in #1010 then.